src/lib/models/outoforder/Core.cc

```diff
@@ -351,11 +351,11 @@ void Core::flushIfNeeded() {
     // Then empty the buffers
     fetchUnit_.flushLoopBuffer();
     fetchUnit_.updatePC(targetAddress);
-    fetchToDecodeBuffer_.flushBranchMacroOps(branchPredictor_);
+    branchPredictor_.flushBuffer(fetchToDecodeBuffer_);
     fetchToDecodeBuffer_.fill({});
```

Comment:
I think this is a decent solution to keeping the pipeline buffers agnostic to their elements. I think the name misconstrues the functionality within the BP here though. This isn't "flushing the buffer" as I first thought, that is happening on the lines below, which confused me for a little while. I think the name should convey the fact that the BP itself is being updated rather than the buffer, but I can't think of anything good. `combBufferAndFlushSelf` `flushAccordingToBuffer` `flushWithBuffer`